--fix converts to backtick, then valid-typeof reports error

**Tell us about your environment**

* **ESLint Version:** 3.5.0
* **Node Version:** 6.5.0
* **npm Version:** 3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?** "babel-eslint": "6.1.2"

**Please show your full configuration:**
.eslintrc.json:

```json
{
  "parser": "babel-eslint",
  "extends": "airbnb",
  "env": {
    "node": true
  },
  "globals": {
    "emit": true,
    "doc": true
  },
  "plugins": [
      "react"
  ],
  "rules": {
    "react/require-extension": "off",
    "linebreak-style": ["error", "windows"],
    "react/jsx-sort-props": 0,
    "react/jsx-no-bind": 0,
    "comma-dangle": 0,
    "no-use-before-define": 0,
    "consistent-return": 0,
    "semi": [2, "never"],
    "quotes": ["error", "backtick"],
    "strict": 0,
    "max-len": ["error", 120, 2, {"ignoreUrls": true}],
    "no-param-reassign": ["error", { "props": false }]
  }
}
```

package.json:
```json
{
  "name": "apflora",
  "version": "1.0.1",
  "description": "Aktionsplan Flora Kt. Zürich Backend",
  "main": "server.js",
  "dependencies": {
    "ampersand-app": "^2.0.0",
    "async": "2.0.1",
    "dateformat": "1.0.12",
    "hapi": "15.0.3",
    "hapi-node-postgres": "4.0.1",
    "inert": "4.0.2",
    "json2csv": "3.6.3",
    "json2xls": "0.1.2",
    "lodash": "4.15.0",
    "pg": "6.1.0",
    "pg-promise": "^5.3.3",
    "request": "2.74.0",
    "underscore": "1.8.3"
  },
  "devDependencies": {
    "babel-eslint": "6.1.2",
    "code": "3.0.2",
    "del": "2.2.2",
    "eslint": "3.5.0",
    "eslint-config-airbnb": "11.1.0",
    "eslint-plugin-import": "1.15.0",
    "eslint-plugin-jsx-a11y": "2.2.2",
    "eslint-plugin-react": "6.2.1",
    "gulp": "3.9.1",
    "gulp-sftp": "0.1.5",
    "lab": "11.0.1",
    "require-dir": "0.3.0",
    "run-sequence": "1.2.2"
  },
  "scripts": {
    "start": "npm test && node startServer.js",
    "startWithoutTests": "node startServer.js",
    "predeploy": "gulp prod_clean_dist",
    "deploy": "gulp prod_copy",
    "postdeploy": "gulp prod_sftp",
    "test": "eslint src --fix && lab",
    "yolo": "npm run deploy"
  },
  "repository": {
    "type": "git",
    "url": "git://github.com/FNSKtZH/apflora_api.git"
  },
  "keywords": [
    "apflora",
    "aktionsplan",
    "flora",
    "species",
    "conservation"
  ],
  "author": "Alexander Gabriel",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/FNSKtZH/apflora/issues"
  },
  "homepage": "https://github.com/FNSKtZH/apflora"
}

```

**What did you do? Please include the actual source code causing the issue.**

I added

`"quotes": ["error", "backtick"]`

to .eslintrc and

`eslint src --fix` to the scripts in package.json

after reading [this](https://ponyfoo.com/articles/template-literals-strictly-better-strings)

**What did you expect to happen?**

`'` to be replaced by backtips ONLY where applicable (as like: not in `'use strict'`).
OR: Other rules not to complain about where `--fix` sets backtips.

**What actually happened? Please include the actual, raw output from ESLint.**

this code:

```
'use strict'

module.exports = (string) => {
  if (string && typeof string === `string`) {
    // does something
  }
}
```

produced this error:

    C:\Users\alex\apflora_api\src\removeKmlNogoStrings.js
      10:35   error  Typeof comparisons should be to string literals  valid-typeof

--- a/lib/rules/valid-typeof.js
+++ b/lib/rules/valid-typeof.js
@@ -58,8 +58,10 @@ module.exports = {
                     if (parent.type === "BinaryExpression" && OPERATORS.indexOf(parent.operator) !== -1) {
                         const sibling = parent.left === node ? parent.right : parent.left;
 
-                        if (sibling.type === "Literal") {
-                            if (VALID_TYPES.indexOf(sibling.value) === -1) {
+                        if (sibling.type === "Literal" || sibling.type === "TemplateLiteral" && !sibling.expressions.length) {
+                            const value = sibling.type === "Literal" ? sibling.value : sibling.quasis[0].value.cooked;
+
+                            if (VALID_TYPES.indexOf(value) === -1) {
                                 context.report(sibling, "Invalid typeof comparison value.");
                             }
                         } else if (requireStringLiterals && !isTypeofExpression(sibling)) {

--- a/tests/lib/rules/valid-typeof.js
+++ b/tests/lib/rules/valid-typeof.js
@@ -59,6 +59,20 @@ ruleTester.run("valid-typeof", rule, {
         {
             code: "typeof foo === typeof bar",
             options: [{ requireStringLiterals: true }]
+        },
+        {
+            code: "typeof foo === `string`",
+            options: [{ requireStringLiterals: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "`object` === typeof foo",
+            options: [{ requireStringLiterals: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "typeof foo === `str${somethingElse}`",
+            parserOptions: { ecmaVersion: 6 }
         }
     ],
 
@@ -112,6 +126,11 @@ ruleTester.run("valid-typeof", rule, {
             errors: [{ message: "Invalid typeof comparison value.", type: "Literal" }]
         },
         {
+            code: "if (typeof bar === `umdefined`) {}",
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ message: "Invalid typeof comparison value.", type: "TemplateLiteral" }],
+        },
+        {
             code: "typeof foo == 'invalid string'",
             options: [{ requireStringLiterals: true }],
             errors: [{ message: "Invalid typeof comparison value.", type: "Literal" }]
@@ -135,6 +154,18 @@ ruleTester.run("valid-typeof", rule, {
             code: "undefined == typeof foo",
             options: [{ requireStringLiterals: true }],
             errors: [{ message: "Typeof comparisons should be to string literals.", type: "Identifier" }]
+        },
+        {
+            code: "typeof foo === `undefined${foo}`",
+            options: [{ requireStringLiterals: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ message: "Typeof comparisons should be to string literals.", type: "TemplateLiteral" }]
+        },
+        {
+            code: "typeof foo === `${string}`",
+            options: [{ requireStringLiterals: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ message: "Typeof comparisons should be to string literals.", type: "TemplateLiteral" }]
         }
     ]
 });

